ctapipe/instrument/trigger.py

```diff
@@ -0,0 +1,97 @@
+from ctapipe.containers import ArrayEventContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import Integer, IntTelescopeParameter
+
+
+class SoftwareTrigger(TelescopeComponent):
+    """
+    A stereo trigger that can remove telescope events from subarray events.
+
+    This class is needed to correctly handle super-arrays as simulated for
+    CTA and still handle the LST hardware stereo trigger and the normal stereo
+    trigger correctly.
+
+    LSTs have a hardware trigger, that requires at least two LSTs to have a trigger
+    in each event. Thus, after selecting a subarray, it is needed to
+    - Remove single LST telescope events from events
+    - Ignore events with only 1 telescope after this has been applied
+    """
+
+    min_telescopes = Integer(
+        default_value=1,
+        help=(
+            "Minimum number of telescopes required globally."
+            " Events with fewer telescopes will be filtered out completely."
+        ),
+    ).tag(config=True)
+
+    min_telescopes_of_type = IntTelescopeParameter(
+        default_value=1,
+        help=(
+            "Minimum number of telescopes required for a specific type."
+            " In events with fewer telescopes of that type"
+            " , those telescopes will be removed from the array event."
+            " This might result in the event not fullfilling ``min_telescopes`` anymore"
+            " and thus being filtered completely."
+        ),
+    ).tag(config=True)
+
+    def __init__(self, subarray, *args, **kwargs):
+        super().__init__(subarray, *args, **kwargs)
+
+        self._ids_by_type = {
+            str(type): set(self.subarray.get_tel_ids_for_type(type))
+            for type in self.subarray.telescope_types
+        }
+
+        print("SOFTWARE_TRIGGER", self.min_telescopes, self.min_telescopes_of_type)
+
+    def __call__(self, event: ArrayEventContainer) -> bool:
+        """
+        Remove telescope events that have not the required number of telescopes of
+        a given type from the subarray event and decide if the event would
+        have triggered the stereo trigger.
+
+        Data is cleared from events that did not trigger.
+
+        Returns
+        -------
+        triggered : bool
+            Whether or not this event would have triggered the stereo trigger
+        """
+
+        for tel_type in self.subarray.telescope_types:
+            tel_type_str = str(tel_type)
+            min_tels = self.min_telescopes_of_type.tel[tel_type_str]
+
+            # no need to check telescopes for which we have no min requirement
+            if min_tels == 1:
```

Comment:
I think the default should be `0`, one could require a single MST for example

Reply:
ok, yes